tests/Athena.NET.Tests/Lexing/TokenReaderTestsBase.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Athena.NET.Lexing;
+using Athena.NET.Lexing.LexicalAnalysis;
+using Athena.NET.Lexing.Structures;
+using System.Text;
+
+namespace Athena.NET.Tests.Lexing;
+
+public class TokenReaderTestsBase
+{
+    protected static Token[] Tokens(params (TokenIndentificator, string)[] declarations)
+    {
+        return declarations
+            .Select(d => new Token(d.Item1, d.Item2.AsMemory()))
+            .ToArray();
+    }
+
+    protected static Token[] Tokenize(string text)
+    {
+        var stream = new MemoryStream();
+        stream.Write(Encoding.UTF8.GetBytes(text));
+        stream.Position = 0;
+
+        using var tokenReader = new TokenReader(stream);
+        return tokenReader
+            .ReadTokensAsync()
+            .GetAwaiter()
+            .GetResult()
+            .ToArray();
+    }
+
+    private protected static TokenReader CreateReader(string text)
```

Comment:
> I don't see any implementation

This is the implementation, you probably meant to say is that it is not referenced anywhere, which is true. It is there for future use. You can remove it, if you wish, but there are many unused methods in your codebase, so it didn't seem to matter.

Reply:
Of course my codebase have many unused methods, however there are not any use cases for it yet. Whereas in your `TokenReaderTestsBase` it could be called in method `Tokenize`, because your first three lines are exactly same.